### DISCUSSION TITLE
How to properly set field as array value?

### DISCUSSION THREAD

**Discussion by deanCompRSA** [vueform/vueform#279] [Questions] [answered]
I have a custom grid component that is a table with various columns. The component renders fine however in the form data the field for it is not present, but just an empty object. I have a modal form that should add the data as an object to the employment array in the form.

My schema is as follows:
`{
    employment: {
        "type": "custom-grid",
        "field": "employment",
        "label": "Work History",
        "rules": ["array"],
        "columns": [
            {
                "columnField": "title",
                "label": "Your Title",
            },
            {
                "columnField": "organization",
                "label": "Company Name",
            },
            {
                "columnField": "position",
                "label": "Is this your current position?",
            },
            {
                "columnField": "contactpreferredname",
                "label": "Supervisor's Name",
            },
            {
                "columnField": "scontacttitle",
                "label": "Supervisor's Title",
            },
            {
                "columnField": "contactEmailAddress",
                "label": "Employment - Contact Email Address",
            }
        ],
        "default": []
    }
}
`

But when I log my form data it is just an empty object. I was expecting {subjectemployment: []} if no rows have been added. If I use a TextElement instead the form properly logs {subjectemployment: ''}. So I can't add to the subjectemployment as it does not exist in the form data. Strangely enough though, any  rules I set up such as min for the array still validates fine.

How do I set up my custom component value to be treated as an array of objects? If someone could show a simple example please

**Comment by adamberecz** [comment]
As a generic advice I would recommend checking out this:
https://vueform.com/docs/creating-elements#nullvalue

To help you with the implementation (which I'm happy to do so), please create a sandbox with your custom element (or some form of minimal representation of it): https://stackblitz.com/github/vueform/sandbox

**Comment by deanCompRSA** [comment]
Here you go https://stackblitz.com/edit/github-nzqvai?file=src%2FApp.vue

You will notice when a row is added, it is not added under the field name employment, where employment is suppose to be an array. But instead it adds the row to the form as 
```
{
  "0": {
    "title": "My Title",
    "organization": "My Org",
    "position": "My position",
    "contactpreferredname": "My contact",
    "contacttitle": "My Contact Title",
    "contactEmailAddress": "My contact email"
  }
}
```

 What I'm expecting is to apply the value to the field name like it does for hello_world, but as part of it's array
 
```
 {
   employment: [
   {
      "title": "My Title",
      "organization": "My Org",
      "position": "My position",
      "contactpreferredname": "My contact",
      "scontacttitle": "My Contact Title",
      "contactEmailAddress": "My contact email"
    }
  ]
}
```

**Comment by adamberecz** [comment, marked as answer]
Instead of `data` you should change `value`:
https://stackblitz.com/edit/github-nzqvai-v14ns1?file=src%2Fcomponents%2FCustomGridElement.vue

The `value` represents the actual value of the element, the [`data`](https://github.com/vueform/vueform/blob/main/src/composables/elements/useData.js#L57) is what gets [merged](https://github.com/vueform/vueform/blob/main/src/composables/useVueform.js#L502) into form data.

**Reply by deanCompRSA** [reply]
Thank you!
